### PR TITLE
LSPS0: Clarify JSON-RPC error codes

### PR DESCRIPTION
**Pull request by SeverinAlexB**
As discussed in #86, LSPS0 was confusing on error codes that were not specifically mention in LSPS0.

This PR adds clarifying sentences.

CC @tnull

### CODE REVIEW

**Comment by SeverinAlexB** [comment]
Updated the PR to include error code ranges for each LSPS.

I did give each LSPS 100 error codes instead of 1000 discussed yesterday. I personally find 1000 to be very excessive in the context that we barely use custom codes in the specs so far:
- LSPS1: 3 codes
- LSPS2: 3 codes
- LSPS4: None defined so far
- LSPS5: 3 codes so far

Personally, I rather have spare error codes than run out of codes. Happy to change if you think otherwise.

File: LSPS0/README.md
Comment (SeverinAlexB) [reply LSPS0/README.md:452]: I can. Can you explain how a client should handle such an error?

From what I can read, a LSP throwing a `-32000 to -32099` error should be handled similar to `-32603 | Internal error`. Or how should a client react?

Comment (SeverinAlexB) [reply LSPS0/README.md:452]: Added section as discussed in todays meeting.

**Comment by SeverinAlexB** [comment]
Update: 
- Added clarifying section for the `-32000 to -32099` error code range.
- Updated the LSPS1 & LSPS2 error codes.

File: LSPS2/README.md
Comment (ErikDeSmedt) [inline LSPS2/README.md:585]: Nit: We use the error-code 200-twice.

I think we have enough space so we don't have re-use numbers. This makes it a bit simpler to implement a response-parser.

File: LSPS1/README.md
Comment (ErikDeSmedt) [inline LSPS1/README.md:238]: Should we move this to LSPS0?

I think this option is useful in every LSPS. 
Especially LSPS2 where a server might have opened multiple channels for which the client never paid.

**Review by ErikDeSmedt — APPROVED** [review]

File: LSPS1/README.md
Comment (SeverinAlexB) [reply LSPS1/README.md:238]: We could add common errors to the LSPS0 Common Schemas doc.

File: LSPS2/README.md
Comment (SeverinAlexB) [reply LSPS2/README.md]: Done

**Comment by SeverinAlexB** [comment]
Does LSPS2 need a `Client rejected` error too?

**Comment by johncantrell97** [comment]
> > Does LSPS2 need a `Client rejected` error too?
> 
> It would be pretty helpful, I think. I know some early adopters were running into issues because they wanted to restrict access based on the token for testing purposes but there wasn't a good message to communicate that. So they just stopped responding which isn't nice.

Yes, there _is_ already an error message for invalid token but we would like to have a way to handle errors due to our internal rate limiting and then of course "internal server error" equivalent when something unexpected goes wrong.

**Comment by SeverinAlexB** [comment]
>Yes, there is already an error message for invalid token but we would like to have a way to handle errors due to our internal rate limiting 

PR for `client_rejected` error -> https://github.com/BitcoinAndLightningLayerSpecs/lsp/pull/102

> and then of course "internal server error" equivalent when something unexpected goes wrong.

`-32603 Internal error` should be the way to go here. See https://www.jsonrpc.org/specification#error_object.

**Comment by SeverinAlexB** [comment]
Update: Client rejected error is now `001` instead of `1`. Waiting for @JssDWt to do the last review before I merge.

File: LSPS1/README.md
Comment (JssDWt) [inline LSPS1/README.md:238]: Nit: let's use alphabetical ordering. 001 before 100.

**Review by JssDWt — APPROVED** [review]
A small nit, otherwise approved. Looking good.

**Comment by SeverinAlexB** [comment]
@JssDWt approved this PR. NIT is fixed. As discussed in our last meeting, I am merging this PR now.

**Pull request merged**
